lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -250,9 +265,12 @@ export abstract class ClientApplication {
             this.eventHandler.emitEvent(EventType.LOGIN_START, InteractionType.Popup, request);
         }
 
-        const popupClient = new PopupClient(this.config, this.browserStorage, this.browserCrypto, this.logger, this.eventHandler, this.navigationClient, request.correlationId);
+        const popupClient = new PopupClient(this.config, this.browserStorage, this.browserCrypto, this.logger, this.eventHandler, this.navigationClient, this.performanceClient, correlationId);
 
-        return popupClient.acquireToken(request).then((result) => {
+        return popupClient.acquireToken({
+            ...request,
+            correlationId
```

Comment:
nit: So there is no central place to retrieve correlation id? Isn't it a part of the request? Or is this the distinction you were making in the discussion?

Reply:
`correlationId` might be passed by the application on the request, might no. Since we want to ensure it is set for the performance telemetry, I added `getRequestCorrelationId` to ensure that we either use the correlation id from the application or generate a new one at the start of the operation.